### ISSUE TITLE
Preserve Scroll Position when we use appendRows

### ISSUE THREAD

**Issue by chrissmartin** [nhn/tui.grid#1298]
## Version

"@toast-ui/vue-grid": "^4.15.1",

## Development Environment

Chrome Version 89.0.4389.90
Mac OS 11.8.3
Using this in Vue with Typescript
## Current Behavior

I'm trying to add a custom implementation of infinite scroll, when we scroll to end of table, I trigger an event which inturn calls the fetch fn fetching next set of items. Then I append these rows using `gridInstance.invoke('appendRows', rowData)`. Everything works without issues, only thing is, after appending the rows the scroll position resets to top. I want to keep the scroll position and just append data to bottom giving a overall smooth experience. 

Right now, I just use the `focusAt` function to scroll to last position again after fetching data. But this is janky.

**Comment by js87zz** [comment]
@chrissmartin 
For example, if there is a grid API that scrolls to the top, such as `scrollToTop()`, it can be useful in this case. I'll consider it. Thank you for your opinion!

**Comment by chrissmartin** [comment]
> @chrissmartin
> For example, if there is a grid API that scrolls to the top, such as `scrollToTop()`, it can be useful in this case. I'll consider it. Thank you for your opinion!

Yup that does make sense. Thou I feel, just keeping the scroll position at the same position makes the user feel, that the data is loaded and which rows are new and all. 
Also, on side note, if there is like a proper infinite scroll much like how instagram does, where in the table shows like a loader when we scroll past last item and if only user like pulls/scrolls till end of this loader animation will the table trigger the event. And when the data is being fetched from API, if there's a loader at center bottom it would be really nice.

Currently I just use a custom scrollEnd event which is not that smooth or working like 100% of time and when loading data, I just show a small custom loader over the table.

**Comment by ShinEunYoung** [comment]
제목을 보고 해당 기능이 저도 필요해서 글을 읽게 되었는데, 혹시 해당 검토중이시라면 해당 기능이 꼭 최상단이나 최하단이 아닌 현재 스크롤 위치를 유지하는 기능으로 개발되었으면 합니다. 혹시 이와 관련한 개발이 언제쯤 업데이트 될 예정인지 알 수 있을까요? 저같은 경우에는 resetData 를 이용해 데이터를 업데이트 할 때 스크롤 이슈가 발생합니다. 혹시 현재 화면 스크롤을 유지하면서 data의 값을 바꿔줄 수 있는 방법이 있다면 공유부탁드립니다

**Comment by treuga** [comment]
> 제목을 보고 해당 기능이 저도 필요해서 글을 읽게 되었는데, 혹시 해당 검토중이시라면 해당 기능이 꼭 최상단이나 최하단이 아닌 현재 스크롤 위치를 유지하는 기능으로 개발되었으면 합니다. 혹시 이와 관련한 개발이 언제쯤 업데이트 될 예정인지 알 수 있을까요? 저같은 경우에는 resetData 를 이용해 데이터를 업데이트 할 때 스크롤 이슈가 발생합니다. 혹시 현재 화면 스크롤을 유지하면서 data의 값을 바꿔줄 수 있는 방법이 있다면 공유부탁드립니다

저도 기능 기다리다가 결국 직접 만들어서 썼는지라 혹시 도움이 될까 남겨둡니다 
- 스크롤 위치 저장할 전역 변수 myLastScroll 만듦
- tui-grid.js 내부 setScrollTop 함수에 myLastScroll = scrollTop; 추가하여 전역변수에 스크롤 위치 저장
- 그리드 페이지에서 row추가나 리셋데이타 후 grid.dispatch('setScrollTop', myLastScroll) 하면 해당 스크롤을 유지하며 데이터를 다시 호출할 수 있습니다

**Comment by chrissmartin** [comment]
> > 제목을 보고 해당 기능이 저도 필요해서 글을 읽게 되었는데, 혹시 해당 검토중이시라면 해당 기능이 꼭 최상단이나 최하단이 아닌 현재 스크롤 위치를 유지하는 기능으로 개발되었으면 합니다. 혹시 이와 관련한 개발이 언제쯤 업데이트 될 예정인지 알 수 있을까요? 저같은 경우에는 resetData 를 이용해 데이터를 업데이트 할 때 스크롤 이슈가 발생합니다. 혹시 현재 화면 스크롤을 유지하면서 data의 값을 바꿔줄 수 있는 방법이 있다면 공유부탁드립니다
> 
> 저도 기능 기다리다가 결국 직접 만들어서 썼는지라 혹시 도움이 될까 남겨둡니다
> 
> * 스크롤 위치 저장할 전역 변수 myLastScroll 만듦
> * tui-grid.js 내부 setScrollTop 함수에 myLastScroll = scrollTop; 추가하여 전역변수에 스크롤 위치 저장
> * 그리드 페이지에서 row추가나 리셋데이타 후 grid.dispatch('setScrollTop', myLastScroll) 하면 해당 스크롤을 유지하며 데이터를 다시 호출할 수 있습니다

- I use the same kinda approach but using `focusAt` fn.  
`this.getGrid()?.invoke('focusAt', newPosition, 0, true);`

The only issue is, it is not preserving the scroll, it does scroll after the data is fetched, which feels a bit janky. But the focus at helps to highlight the row of the last end.
